src/main/java/org/jenkinsci/plugins/github_branch_source/RepositoryUriResolver.java

```diff
@@ -38,17 +40,14 @@
 
     public static String hostnameFromApiUri(String apiUri) {
         apiUri = Util.fixEmptyAndTrim(apiUri);
-        if (apiUri == null) {
-            return "github.com";
+        try {
+            URL endpoint = new URL(apiUri);
```

Comment:
IMO, everything will be easier if we replace [this line](https://github.com/jenkinsci/github-branch-source-plugin/blob/master/src/main/java/org/jenkinsci/plugins/github_branch_source/AbstractGitHubSCMSource.java#L297) for: ```result.add("Github", "https://github.com");```